flink-table/flink-table-planner-blink/src/main/scala/org/apache/flink/table/planner/codegen/calls/ScalarOperatorGens.scala

```diff
@@ -1643,19 +1644,35 @@ object ScalarOperatorGens {
     val resultTypeTerm = primitiveTypeTermForType(componentInfo)
     val defaultTerm = primitiveDefaultValue(componentInfo)
 
+    if (index.literalValue.isDefined &&
+        index.literalValue.get.isInstanceOf[Int] &&
+        index.literalValue.get.asInstanceOf[Int] < 1) {
+      throw new ValidationException(s"Array element access needs an index starting at 1 but was " +
```

Comment:
If we cannot do this in calcite's parsing/validating phase, it's ok to me to check it here.

Reply:
"Array element reference requires an index starts from 1, but was"

Reply:
> "Array element reference requires an index starts from 1, but was"
This text is same to [existed Table API validation](https://github.com/apache/flink/blob/master/flink-table/flink-table-planner/src/main/scala/org/apache/flink/table/expressions/collection.scala#L212) so that we can keep align.